python_modules/libraries/dagster-dg/dagster_dg/utils/__init__.py

```diff
@@ -311,10 +318,13 @@ def hash_directory_metadata(
 
 
 def hash_file_metadata(hasher: Hash, path: Union[str, Path]) -> None:
-    stat = os.stat(path=path)
-    hasher.update(str(path).encode())
-    hasher.update(str(stat.st_mtime).encode())  # Last modified time
-    hasher.update(str(stat.st_size).encode())  # File size
+    try:
+        stat = os.stat(path=path)
+        hasher.update(str(path).encode())
+        hasher.update(str(stat.st_mtime).encode())  # Last modified time
+        hasher.update(str(stat.st_size).encode())  # File size
+    except FileNotFoundError:
+        pass
```

Comment:
should we really just swallow this error silently in all cases? At least comment-worthy

Reply:
+1, add a warning? or maybe a flag that is called by the watcher if its expected for this to happen while files are getting moved around while things are being checked

Reply:
updated to add a flag, we'll only ignore the error for changes that the watch handler encounters (when the filesystem might be actively changing)